xrdp/xrdp_wm.c

```diff
@@ -1775,13 +1798,31 @@ xrdp_wm_process_input_mouse(struct xrdp_wm *self, int device_flags,
     /* vertical mouse wheel */
     if (device_flags & PTRFLAGS_WHEEL)
     {
+        int delta = 0;
         if (device_flags & PTRFLAGS_WHEEL_NEGATIVE)
         {
-            xrdp_wm_mouse_click(self, 0, 0, 5, 0);
+            // [MS-RDPBCGR] In negative scrolling, rotation distance is negative.
+            delta = (device_flags & WheelRotationMask) | ~WheelRotationMask;
```

Comment:
Can you add a reference to 2.2.8.1.1.3.1.1.3 here? It took me a while to figure out what line 1805 was doing and I needed the reference.

Reply:
👌I'll add more comments and the reference link.

Reply:
After some additional changes, I'll have a look.

Reply:
ok. done.